src/tmpfiles/tmpfiles.c

```diff
@@ -1261,7 +1261,9 @@ static int create_item(Item *i) {
                 log_debug("%s directory \"%s\".", creation_mode_verb_to_string(creation), i->path);
 
                 if (IN_SET(i->type, CREATE_SUBVOLUME_NEW_QUOTA, CREATE_SUBVOLUME_INHERIT_QUOTA)) {
-                        r = btrfs_subvol_auto_qgroup(i->path, 0, i->type == CREATE_SUBVOLUME_NEW_QUOTA);
+                        r = path_check_fstype(i->path, BTRFS_SUPER_MAGIC);
+                        if (r == 1)
+                                 r = btrfs_subvol_auto_qgroup(i->path, 0, i->type == CREATE_SUBVOLUME_NEW_QUOTA);
```

Comment:
note that a check like this is actually done inside of btrfs_subvol_auto_qgroup() anyway, as we don't want to issue btrfs ioctls on fds that are not btrfs, since ioctls are after all not assigned uniquely. The only difference appears to be that btrfs_subvol_auto_qgroup() operates on a normal fd, while  path_check_fstype() operates on an O_PATH one...

I wonder if we could adjust  btrfs_subvol_auto_qgroup() to use an O_PATH fd, too? Not sure if the btrfs ioctls would work on an O_PATH fd though...

What are the precise implications of O_PATH on autofs mounts?

Hmm, I am also a bit concerned about the fact that this check and the btrfs operation are two independent steps... This is racy after all, and I'd feel more comfortable if there was a way to somehow do all this with the same fd or so, so that we at least know we are talking about the same file system object the whole time...